src/Estimators/tests/test_InputSection.cpp

```diff
@@ -0,0 +1,335 @@
+</test>
+)"},
+        {"foreign_parameter", R"(
+<test full="no">
+  <parameter name="count"> 15 </parameter>
+  <parameter name="area" > 51 </parameter>
+</test>
+)"},
+    };
+
+    for (auto& [label, xml] : invalid_inputs)
+    {
+      // parse xml doc
+      Libxml2Document doc;
+      bool okay = doc.parseFromString(xml);
+      REQUIRE(okay);
+      xmlNodePtr cur = doc.getRoot();
+
+      // read xml
+      TestInput ti;
+      //app_log()<<"trying xml read for "<<label<<"\n";
+      try
+      {
+        ti.readXML(cur);
+      }
+      catch (std::runtime_error& exception)
+      {
+        //app_log()<<label<<" is caught correctly\n";
+      }
+    }
+  }
+}
+
+
+TEST_CASE("InputSection::init", "[estimators]")
+{
+  { // test init: minimum required attributes and parameters
+    // initialize
+    TestInput ti;
+    ti.init({{"full", false}, {"count", 15}});
+
+    // assigned from initializer-list
+    CHECK(ti.has("full"));
+    CHECK(ti.has("count"));
+    // assigned from defaults
+    CHECK(ti.has("name"));
+    CHECK(ti.has("samples"));
+    CHECK(ti.has("width"));
+    CHECK(ti.has("rational"));
+    // unassigned
+    CHECK(!ti.has("kmax"));
+    CHECK(!ti.has("label"));
+    // check value correctness
+    CHECK(ti.get<bool>("full") == false);
+    CHECK(ti.get<int>("count") == 15);
+    CHECK(ti.get<std::string>("name") == "demo");
+    CHECK(ti.get<int>("samples") == 20);
+    CHECK(ti.get<RealType>("width") == Approx(1.0));
+    CHECK(ti.get<bool>("rational") == false);
+  }
+
+
+  { // test init: complete attributes and parameters
+    // initialize
+    TestInput ti;
+    ti.init({
+        {"name", std::string("alice")},
+        {"samples", 10},
+        {"kmax", 3.0},
+        {"full", false},
+        {"label", std::string("relative")},
+        {"count", 15},
+        {"width", 2.5},
+        {"rational", true},
+    });
+
+    // assigned from initializer-list
+    CHECK(ti.has("name"));
+    CHECK(ti.has("samples"));
+    CHECK(ti.has("kmax"));
+    CHECK(ti.has("full"));
+    CHECK(ti.has("label"));
+    CHECK(ti.has("count"));
+    CHECK(ti.has("width"));
+    CHECK(ti.has("rational"));
+    // check value correctness
+    CHECK(ti.get<std::string>("name") == "alice");
+    CHECK(ti.get<int>("samples") == 10);
+    CHECK(ti.get<RealType>("kmax") == Approx(3.0));
+    CHECK(ti.get<bool>("full") == false);
+    CHECK(ti.get<std::string>("label") == "relative");
+    CHECK(ti.get<int>("count") == 15);
+    CHECK(ti.get<RealType>("width") == Approx(2.5));
+    CHECK(ti.get<bool>("rational") == true);
+  }
+}
+
+
+TEST_CASE("InputSection::get", "[estimators]")
+{
+  TestInput ti;
+  ti.init({
+      {"name", std::string("alice")},
+      {"samples", 10},
+      {"kmax", 3.0},
+      {"full", false},
+      {"label", std::string("relative")},
+      {"count", 15},
+      {"width", 2.5},
+      {"rational", true},
+  });
+
+  // invalid type access results in thrown exception
+  try
```

Comment:
use CHECK_THROWS or CHECK_THROWS_AS

Reply:
As is I'm not sure what this actually tests since I'm pretty sure it can't fail.

Reply:
No, it can't based on the current underlying implementation.  It should be tested in case that implementation changes for some reason down the road.